### ISSUE TITLE
[JENKINS-69273] `ignoreFailedBuilds: false` does not show charts without at least one passed build 

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/warnings-ng-plugin#3095]
<p>This is a follow-up issue, as suggested by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=drulli" class="user-hover" rel="drulli">drulli</a>: <a href="https://issue-redirect.jenkins.io/issue/59960" title="no graph trend in multi branch job" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/59960">JENKINS-59960</a>"><del>JENKINS-59960</del></a></p>

<p> </p>

<p>In short:</p>

<p><em>wngp</em> does not show charts for jobs where all builds are failed (in other words, no passed build), even if `ignoreFailedBuilds` is set to 'false'. </p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/schperplata.jpg" title="schperplata's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=schperplata">schperplata</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-69273" target="_blank">`ignoreFailedBuilds: false` does not show charts without at least one passed build </a></i></summary>
<i><ul>
<li><b>status</b>: Open
<li><b>priority</b>: Minor
<li><b>component(s)</b>: warnings-ng-plugin
<li><b>label(s)</b>: hacktoberfest, help-wanted, newbie-friendly, user-experience, ux
<li><b>resolution</b>: Unresolved
<li><b>votes</b>: 0
<li><b>watchers</b>: 1
<li><b>imported</b>: 2025-12-09
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>This is a follow-up issue, as suggested by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=drulli" class="user-hover" rel="drulli">drulli</a>: <a href="https://issues.jenkins.io/browse/JENKINS-59960" title="no graph trend in multi branch job" class="issue-link" data-issue-key="JENKINS-59960"><del>JENKINS-59960</del></a></p>

<p> </p>

<p>In short:</p>

<p><em>wngp</em> does not show charts for jobs where all builds are failed (in other words, no passed build), even if `ignoreFailedBuilds` is set to 'false'. </p></pre>
</details>
</details>

**Comment by ShashwathNayak06** [comment]
Hi, I’m exploring Jenkins plugin development and have been looking into the warnings-ng plugin. I’d like to investigate this issue and see if I can propose a fix. Please let me know if that would be okay or if there’s any guidance you’d recommend before starting.

**Comment by uhafner** [comment]
Hi @shaszz, yes please go ahead and have a look at this issue. The best starting place is to recreate the issue in you local Jenkins instance. If you can reproduce the issue in your Jenkins then the next step would be to write an integration test, that reproduces the bug in code. An example is in https://github.com/jenkinsci/warnings-ng-plugin/blob/4091f2ec8b5298a3685a2b07b30385d0d8050aba/plugin/src/test/java/io/jenkins/plugins/analysis/warnings/steps/JobActionITest.java#L49.
 
Using that test you can try to fix the logic and propose a PR.

**Comment by ShashwathNayak06** [comment]
@uhafner , Thanks for the guidance!
I’ll follow up with my findings or a PR once I have the test in place.

**Comment by ShashwathNayak06** [comment]
Hi  @uhafner , I was able to recreate the issue locally in Jenkins: after two failed builds with , the trend chart is still hidden.
I’m now working on adding an integration test in JobActionITest to reproduce this behavior in code.

**Comment by ShashwathNayak06** [comment]
Hi @uhafner , 
 quick clarification: I don’t see ignoreFailedBuilds neither in the current UI nor in the code. In the UI it looks like this is now Enable recording for failed builds  (setEnabledForFailure(true)).

With that enabled, I still don’t see the trend chart when all builds are failed ,it only appears once there’s a successful build. Just wanted to confirm this matches the intended behavior for the issue before finalizing the test.

**Comment by uhafner** [comment]
Yes, that is exactly the flag in the issue. When this flag is enabled, then results are recorded and then the trend should appear with these results.
